frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/SelectedNodeInputOutputTab.tsx

```diff
@@ -4,23 +4,34 @@ import { PipelineRunTaskDetails, TaskReferenceMap } from '~/concepts/pipelines/c
 import { getNameAndPathFromTaskRef, getValue } from '~/concepts/pipelines/topology/pipelineUtils';
 import TaskDetailsInputParams from '~/concepts/pipelines/content/pipelinesDetails/taskDetails/TaskDetailsInputParams';
 import TaskDetailsOutputResults from '~/concepts/pipelines/content/pipelinesDetails/taskDetails/TaskDetailsOutputResults';
+import { PipelineRunTaskParam } from '~/k8sTypes';
 
 type SelectedNodeInputOutputTabProps = {
   task: PipelineRunTaskDetails;
   taskReferences: TaskReferenceMap;
+  parameter?: PipelineRunTaskParam[];
 };
 
 const SelectedNodeInputOutputTab: React.FC<SelectedNodeInputOutputTabProps> = ({
   task,
   taskReferences,
+  parameter,
 }) => {
   const params =
     task.params?.map((p) => {
+      const paramValueMatch = p.value.match(/^\$\((params\.([a-zA-Z0-9-_]+))\)/);
```

Comment:
Perhaps you could move the initial check into a separate function, in a similar way to how `getNameAndPathFromTaskRef` is done below?